File: backend/app/Models/Produit.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Attributes\Casts;
use Illuminate\Database\Eloquent\Attributes\Fillable;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;

#[Fillable(['nom', 'unite_mesure', 'categorie', 'actif'])]
#[Casts(actif: 'boolean')]
class Produit extends Model
{
    protected $table = 'produits';

    public function relevesPrix(): HasMany
    {
        return $this->hasMany(RelevePrix::class);
    }
}

File: frontend/lib/models/produit.dart
class Produit {
  const Produit({
    required this.id,
    required this.nom,
    required this.uniteMesure,
    this.categorie,
    required this.actif,
  });

  final int id;
  final String nom;
  final String uniteMesure;
  final String? categorie;
  final bool actif;

  factory Produit.fromJson(Map<String, dynamic> json) {
    return Produit(
      id: json['id'] as int,
      nom: json['nom'] as String? ?? '',
      uniteMesure: json['unite_mesure'] as String? ?? '',
      categorie: json['categorie'] as String?,
      actif: json['actif'] as bool? ?? true,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'nom': nom,
      'unite_mesure': uniteMesure,
      'categorie': categorie,
      'actif': actif,
    };
  }
}